extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/RadiusConnectionService.java

```diff
@@ -0,0 +1,394 @@
+            logger.error("RADIUS client hasn't been set up, yet.");
+            logger.debug("We can't run this method until the RADIUS client has been set up.");
+            return null;
+        }
+
+        String radAuthName;
+        String caFile;
+        String caPassword;
+        String caType;
+        String keyFile;
+        String keyPassword;
+        String keyType;
+        Boolean trustAll;
+        String innerProtocol;
+        LocalEnvironment guacEnv;
+        String basePath;
+
+        // Pull configuration parameters from guacamole.properties
+        try {
+            guacEnv = new LocalEnvironment();
+            basePath = guacEnv.getGuacamoleHome().getAbsolutePath() + '/';
+            radAuthName = confService.getRadiusAuthProtocol();
+            caFile = confService.getRadiusCAFile();
+            caPassword = confService.getRadiusCAPassword();
+            caType = confService.getRadiusCAType();
+            keyFile = confService.getRadiusKeyFile();
+            keyPassword = confService.getRadiusKeyPassword();
+            keyType = confService.getRadiusKeyType();
+            trustAll = confService.getRadiusTrustAll();
+            innerProtocol = confService.getRadiusEAPTTLSInnerProtocol();
+            
+        }
+        catch (GuacamoleException e) {
+            logger.error("Error retrieving configuration.");
+            logger.debug("Error getting config parameters from file.");
+            return null;
+        }
+
+        RadiusAuthenticator radAuth = radiusClient.getAuthProtocol(radAuthName);
+        if (radAuth == null)
+            return null;
+
+        // If we're using any of the TLS protocols, we need to configure them
+        if (radAuth instanceof PEAPAuthenticator || 
+            radAuth instanceof EAPTLSAuthenticator || 
+            radAuth instanceof EAPTTLSAuthenticator) {
+
+            if (caFile != null && !caFile.isEmpty())
+                ((EAPTLSAuthenticator)radAuth).setCaFile(basePath + caFile);
+
+            if (caType != null && !caType.isEmpty())
+                ((EAPTLSAuthenticator)radAuth).setCaFileType(caType);
+
+            if (caPassword != null && !caPassword.isEmpty())
+                ((EAPTLSAuthenticator)radAuth).setCaPassword(caPassword);
+
+            if (keyFile != null && !keyFile.isEmpty())
+                ((EAPTLSAuthenticator)radAuth).setKeyFile(basePath + keyFile);
+
+            if (keyType != null && !keyType.isEmpty())
+                ((EAPTLSAuthenticator)radAuth).setKeyFileType(keyType);
+
+            if (keyPassword != null && !keyPassword.isEmpty())
+                ((EAPTLSAuthenticator)radAuth).setKeyPassword(keyPassword);
+
+            ((EAPTLSAuthenticator)radAuth).setTrustAll(trustAll);
+
+        }
+
+        // If we're using EAP-TTLS, we need to define tunneled protocol
+        if (radAuth instanceof EAPTTLSAuthenticator) {
+
+            if (innerProtocol == null || innerProtocol.isEmpty())
```

Comment:
Same here - I strongly suggest we not do `foo == null || foo.isEmpty()` for all string values. If user's really want to specify an empty value, let them. If the value is bad, we should be logging an error.

Either way, it really should be the job of the configuration service to deal with nullifying blank values, if that's really the behavior we want. For consistency and reliable strictness, I suggest otherwise.